gameFunctions.js

```diff
@@ -0,0 +1,288 @@
+                        console.log(body);
+                        if (err) {
+                            console.log(err);
+                        }
+
+                        if (body.ok) {
+                            count++;
+                            if (count >= players.length) {
+                                resolve();
+                            }
+                        }
+                    });
+                }
+            });
+        });
+    });
+    return response;
+}
+
+const createGroup = (players, channelId) => {
+    players.forEach(player => {
+        request.post({
+            url: 'https://slack.com/api/conversations.invite',
+            json: true,
+            headers: {
+                Authorization: 'Bearer ' + process.env.SLACK_AUTH_TOKEN
+            },
+            body: {
+                channel: channelId,
+                users: player.id
+            }
+        }, (err, res, body) => {
+            console.log(body);
+            if (err) {
+                console.log(err);
+            }
+        });
+    });
+
+}
+
+const getRoleText = (role) => {
+    switch (role) {
+        case 'seer':
+            return 'You are the seer. Each night you get to view one player. You win when all the BGs are dead.';
+        case 'doc':
+            return 'You are the doctor. Each night you get to save one player from a night death. You win when all the BGs are dead.';
+        case 'bg':
+            return 'You are Bad Guy. You win when there are the same number (or more) BGs left alive than GGs.';
+        default:
+            return 'You are just a plane jane villager. You win when all the BGs are dead.';
+    }
+}
+
+const assignRoles = (poll, roles) => {
+    let players = poll.choices.options[0].votes;
+    if (players.length > 1) {
+        var numberBGs = Math.round(players.length * 0.20) || 1;
+        var assignedPlayers = [];
+        while (players.length > 0) {
+
+            if (assignedPlayers.length < numberBGs) {
+                var rng = randomNumber(players.length);
+                assignedPlayers.push({
+                    id: players[rng],
+                    role: 'bg',
+                    dead: false
+                });
+                players.splice(rng, 1);
+            } else if (roles.length) {
+                var rng = randomNumber(players.length);
+                assignedPlayers.push({
+                    id: players[rng],
+                    role: roles[0],
+                    dead: false
+                });
+                players.splice(rng, 1);
+                roles.splice(0, 1);
+            } else {
+                assignedPlayers.push({
+                    id: players.pop(),
+                    role: null,
+                    dead: false
+                });
+            }
+        }
+
+        return assignedPlayers;
+    }
+}
+
+const randomNumber = (max) => {
+    return Math.floor(Math.random() * Math.floor(max));
+}
```

Comment:
I dislike how java-script does random. nothing to refactor here, just complaining

Reply:
sorry?